core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -50,6 +57,37 @@ object ZkUtils extends Logging {
   val EntityConfigPath = "/config"
   val EntityConfigChangesPath = "/config/changes"
 
+  /** true if java.security.auth.login.config is set to some jaas file which has "Client" entry. **/
+
+  val isSecure: Boolean = {
+    val loginConfigurationFile: String = System.getProperty("java.security.auth.login.config")
+    var isSecurityEnabled  = false
+    if (loginConfigurationFile != null && loginConfigurationFile.length > 0) {
+      val config_file: File = new File(loginConfigurationFile)
+      if (!config_file.canRead) {
+        throw new KafkaException("File " + loginConfigurationFile + " cannot be read.")
+      }
+      try {
+        val config_uri: URI = config_file.toURI
+        val login_conf = Configuration.getInstance("JavaLoginConfig", new URIParameter(config_uri))
+        isSecurityEnabled = login_conf.getAppConfigurationEntry("Client") != null
+      } catch {
+        case ex: Exception => {
+          throw new KafkaException(ex)
+        }
+      }
+    }
+    isSecurityEnabled
+  }
+
+  val DefaultAcls: List[ACL] = if (isSecure) {
+    import scala.collection.JavaConversions._
+    (ZooDefs.Ids.CREATOR_ALL_ACL ++ ZooDefs.Ids.READ_ACL_UNSAFE).toList
```

Comment:
Yes. Do you think we should lock this down? I was worried locking this down so only creator can read will end up braking clients that rely on zookeeper to get some information + the data stored in these nodes do not seem like it needs to be locked down but I can be wrong about.